rust/common/cookieless/src/salt_cache.rs

```diff
@@ -0,0 +1,354 @@
+                }
+            };
+
+            // Store it in the cache
+            self.cache.insert(yyyymmdd.to_string(), salt.clone());
+
+            return Ok(salt);
+        }
+
+        // Generate a new salt
+        let mut new_salt = vec![0u8; 16];
+        rand::thread_rng().fill_bytes(&mut new_salt);
+        let new_salt_base64 = general_purpose::STANDARD.encode(&new_salt);
+
+        // Try to set it in Redis with NX (only if it doesn't exist) and with TTL in a single operation
+        match self
+            .redis_client
+            .set_nx_ex(redis_key.clone(), new_salt_base64.clone(), SALT_TTL_SECONDS)
+            .await
+        {
+            Ok(true) => {
+                // Successfully set the key with NX and TTL
+                // Store it in the cache
+                self.cache.insert(yyyymmdd.to_string(), new_salt.clone());
+
+                Ok(new_salt)
+            }
+            Ok(false) => {
+                // Someone else set it, try to get it again
+                let salt_base64_retry = match self.redis_client.get(redis_key).await {
+                    Ok(value) => value,
+                    Err(e) => {
+                        inc(
+                            COOKIELESS_REDIS_ERROR_COUNTER,
+                            &[
+                                ("operation".to_string(), "get_salt_retry".to_string()),
+                                ("day".to_string(), yyyymmdd.to_string()),
+                            ],
+                            1,
+                        );
+                        return Err(SaltCacheError::RedisError(e.to_string()));
+                    }
+                };
+
+                let salt = match general_purpose::STANDARD.decode(salt_base64_retry) {
+                    Ok(s) => s,
+                    Err(_) => {
+                        inc(
+                            COOKIELESS_REDIS_ERROR_COUNTER,
+                            &[
+                                ("operation".to_string(), "decode_salt_retry".to_string()),
+                                ("day".to_string(), yyyymmdd.to_string()),
+                            ],
+                            1,
+                        );
+                        return Err(SaltCacheError::SaltRetrievalFailed);
+                    }
+                };
+
+                // Store it in the cache
+                self.cache.insert(yyyymmdd.to_string(), salt.clone());
+
+                Ok(salt)
+            }
+            Err(e) => {
+                inc(
+                    COOKIELESS_REDIS_ERROR_COUNTER,
+                    &[
+                        ("operation".to_string(), "set_salt_nx_ex".to_string()),
+                        ("day".to_string(), yyyymmdd.to_string()),
+                    ],
+                    1,
+                );
+                Err(SaltCacheError::RedisError(e.to_string()))
+            }
+        }
+    }
+
+    /// Clear the entire cache
+    pub fn clear_cache(&self) {
```

Comment:
```suggestion
    pub fn clear(&self) {
```

This is a cache, it's obvious that if you call `clear` you're trying to "clear the cache" :)